xml/System.Diagnostics/DiagnosticListener.xml

```diff
@@ -279,9 +290,14 @@
         <Parameter Name="isEnabled" Type="System.Func&lt;System.String,System.Object,System.Object,System.Boolean&gt;" Index="1" FrameworkAlternate="netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0" />
       </Parameters>
       <Docs>
-        <param name="observer">To be added.</param>
-        <param name="isEnabled">To be added.</param>
-        <summary>To be added.</summary>
+        <param name="observer">A subscriber.</param>
+        <param name="isEnabled">A delegate that filters events based on their name and up to two context object (which can be <see langword="null" />).  
+A particular instrumentation site has the option of calling one or more <see cref="M:System.Diagnostics.DiagnosticListener.IsEnabled" /> overloads  in which it passes the name of the event and up to two other (instrumentation site specific) objects as arguments. If any of these <see cref="M:System.Diagnostics.DiagnosticListener.IsEnabled" /> calls are made then this <paramref name="isEnabled" /> predicate is invoked with passed values (if shorter overloads are used, <see langword="null" /> is passed for missing context objects).
+This gives any particular instrumentation site the ability to pass up to two pieces of information to the subscriber to do sophisticated, efficient filtering. This requires more coupling between the instrumentation site and the subscriber code.
+It is expected that a particular instrumentation site may call different overloads of IsEnabled for the same event, first calling <see cref="M:System.Diagnostics.DiagnosticListener.IsEnabled(System.String)" /> which calls the filter with two <see langword="null" /> context objects. If <paramref name="isEnabled" /> returns <see langword="true" />, it calls again with context objects. The <paramref name="isEnabled" /> filter should be designed with this in mind. 
+Note that the <paramref name="isEnabled" /> predicate is an optional optimization to allow the instrumentation site to avoid setting up the payload and calling <see cref="M:System.Diagnostics.DiagnosticListener.Write(System.String,System.Object)" /> when no subscriber cares about it. In particular the instrumentation site has the option of ignoring the <see cref="M:System.Diagnostics.DiagnosticListener.IsEnabled" /> predicate (not calling it) and simply calling <see cref="M:System.Diagnostics.DiagnosticListener.Write(System.String,System.Object)" />. Thus if the subscriber requires the filtering, it needs to do it itself.
+If this parameter is <see langword="null" />, no filtering is done (all overloads of <see cref="M:System.Diagnostics.DiagnosticSource.IsEnabled" /> return <see langword="true" />).</param>
```

Comment:
```suggestion
If this parameter is <see langword="null" />, no filtering is done (all overloads of <see cref="Overload:System.Diagnostics.DiagnosticSource.IsEnabled" /> return <see langword="true" />).</param>
```

Reply:
I am going to modify this as &lt;xref:Overload:...